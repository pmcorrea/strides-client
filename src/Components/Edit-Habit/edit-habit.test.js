import React from 'react';
import ReactDOM from 'react-dom'
import { BrowserRouter } from 'react-router-dom'
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect'
import EditHabit from './edit-habit';

afterEach(cleanup)

// Snapshot Test
it ('should take a snapshot', () => {
	const { asFragment } = render(
	<BrowserRouter>
			<EditHabit />
	</BrowserRouter>
	)

	expect(asFragment(
		<BrowserRouter>
			<EditHabit />
		</BrowserRouter>)
		).toMatchSnapshot()
})

// Smoke Test
it('renders without crashing', () => {
	const div = document.createElement('div')

	ReactDOM.render(
		<BrowserRouter>
			<EditHabit />
		</BrowserRouter>,
		div
	)
	
	ReactDOM.unmountComponentAtNode(div)
})