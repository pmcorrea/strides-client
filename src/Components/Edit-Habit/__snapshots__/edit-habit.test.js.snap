// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      href="/habit"
    >
      <img
        alt=""
        class="edit-icon add-margin"
        src="back.svg"
      />
    </a>
    <form
      action=""
      class="add-habit-form"
    >
      <p>
        I want to
      </p>
      <input
        name="habit"
        placeholder="meditate"
        type="text"
      />
      <p>
        every
      </p>
      <div>
        <button
          class="circle-button-s"
          style="background-color: white; color: black;"
          type="button"
          value="sunday"
        >
          S
        </button>
        <button
          class="circle-button-s"
          style="background-color: white; color: black;"
          type="button"
          value="monday"
        >
          M
        </button>
        <button
          class="circle-button-s"
          style="background-color: white; color: black;"
          type="button"
          value="tuesday"
        >
          T
        </button>
        <button
          class="circle-button-s"
          style="background-color: white; color: black;"
          type="button"
          value="wednesday"
        >
          W
        </button>
        <button
          class="circle-button-s"
          style="background-color: white; color: black;"
          type="button"
          value="thursday"
        >
          T
        </button>
        <button
          class="circle-button-s"
          style="background-color: white; color: black;"
          type="button"
          value="friday"
        >
          F
        </button>
        <button
          class="circle-button-s"
          style="background-color: white; color: black;"
          type="button"
          value="saturday"
        >
          S
        </button>
      </div>
      <p>
        for 30 days
      </p>
      <button
        type="submit"
      >
        <a
          href="/home"
        >
          Submit
        </a>
      </button>
    </form>
  </div>
</DocumentFragment>
`;
